frontend/src/utils/api-helper.ts

```diff
@@ -0,0 +1,64 @@
+const OK_RESPONSE: number = 200;
+const TOO_MANY_REQUESTS_RESPONSE: number = 429;
+
+export class APIHandledCall {
```

Comment:
honestly this is a bit large of an abstraction. we're not going to need onFail, always, onTooManyRequests etc. What I would do instead is have an `APIHelper` that has `get`, taking a URL, and always doing the same thing for 429 etc.

this is a bit bulky. if possible, this should use an http interceptor for maximum usability: https://stackoverflow.com/a/37252819/926845

Reply:
I agree that this may be too much. The reason I didn't go with an interceptor is because of the bailing part (stop execution of code because no actual result returned). I don't think the interceptor can do that, only redirect completely.

I will try to do a best of both worlds; the interceptor to show 429 warning and a APIHelper with a onsuccess callback. Or perhaps a simple inline check for 200 success response and drop the APIHelper alltogether.

I'll see what I can cook on Wednesday.

Reply:
I'm not sure we need an APIHelper for something like that. A typical HTTP call would be:

- make call
- check if data is present
- if no data, bail

That's how it should currently be anyway. A 429 interceptor is a good idea (that's the idea behind interceptors, is to process data on the outgoing or incoming side, but nothing else)